repo/filtered_log.go

```diff
@@ -0,0 +1,81 @@
+package repo
+
+import (
+	"context"
+	"fmt"
+
+	"go.cryptoscope.co/luigi"
+	"go.cryptoscope.co/luigi/mfr"
+	"go.cryptoscope.co/margaret"
+	refs "go.mindeco.de/ssb-refs"
+)
+
+// FilterFunc works on messages of a FilteredLog. If the func returns true, the log is in the filtered log.
+type FilterFunc func(refs.Message) bool
+
+// NewFilteredLog wraps the passed log into a new one, using the FilterFunc to decide if a message is in the log.
+func NewFilteredLog(b margaret.Log, fn FilterFunc) margaret.Log {
+	return FilteredLog{
+		backing: b,
+		filter:  fn,
+	}
+}
+
+type FilteredLog struct {
```

Comment:
i believe FilteredLog is an essential concept, can you comment its overarching purpose?